All_Relationships/2.candidate-labeler.py

```diff
@@ -47,246 +42,170 @@
-
-# In[ ]:
+sql_statements = [
+    '''
+    SELECT id from candidate
+    WHERE split = 9 and type='compound_disease'
+    ORDER BY RANDOM()
+    LIMIT 50;
+    ''',
+    
+    '''
+    SELECT id from candidate
+    WHERE split = 9 and type='compound_disease'
+    ORDER BY RANDOM()
+    LIMIT 10;
+    ''',
 
+    '''
+    SELECT id from candidate
+    WHERE split = 10 and type='compound_disease'
+    ORDER BY RANDOM()
+    LIMIT 10;
+    '''
+]
 
-from  sqlalchemy.sql.expression import func
-labeler = LabelAnnotator(lfs=list(CG_LFS["CbG_DB"].values()) + 
-                         list(CG_LFS["CbG_TEXT"].values()) +   
-                         list(DG_LFS["DaG_TEXT"].values()))
+spreadsheet_names = {
+    'train': 'data/compound_disease/sentence_labels_train.xlsx',
+    'train_hand_label': 'data/compound_disease/sentence_labels_train_dev.xlsx',
+    'dev': 'data/compound_disease/sentence_labels_dev.xlsx'
+}
 
 
 # In[ ]:
 
 
-def make_sentence_df(candidates):
-    """ 
-    This function creats a dataframe for all candidates (sentences that contain at least two mentions)
-    located in our database.
-    
-    candidates - a list of candidate objects passed in from sqlalchemy
-    """
-    rows = list()
-    for c in tqdm_notebook(candidates):
-        row = OrderedDict()
-        row['candidate_id'] = c.id
-        row['compound'] = c[0].get_span()
-        row['disease'] = c[1].get_span()
-        row['drugbank_id'] = c.Compound_cid
-        row['entrez_gene_id'] = c.Gene_cid
-        row['sentence'] = c.get_parent().text
-        rows.append(row)
-    return pd.DataFrame(rows)
-
-
-# ### Train Set
+for sql, spreadsheet_name in zip(sql_statements, spreadsheet_names.values()):
+    target_cids = [x[0] for x in session.execute(sql)]
+    candidates = (
+        session
+        .query(CandidateClass)
+        .filter(CandidateClass.id.in_(target_cids))
+        .all()
+    )
+    candidate_df = make_sentence_df(candidates)
+    write_candidates_to_excel(candidate_df, spreadsheet_name)
 
-# In[ ]:
 
+# # Develop Label Functions
 
-sql = '''
-SELECT id from candidate
-WHERE split = 9 and type='compound_gene'
-ORDER BY RANDOM()
-LIMIT 50000;
-'''
-target_cids = [x[0] for x in session.execute(sql)]
+# ## Look at potential Candidates
 
+# Use this to look at loaded candidates from a given set. The constants represent the index to retrieve the appropiate set. Ideally, here is where one can look at a subset of the candidate and develop label functions for candidate labeling.
 
 # In[ ]:
 
 
-candidates = session.query(CompoundGene).filter(CompoundGene.id.in_(target_cids)).all()
+train_df = pd.read_excel(spreadsheet_names['train'])
+train_df.head(2)
 
 
 # In[ ]:
 
 
-train_df = make_sentence_df(candidates)
-train_df.head(2)
+train_candidate_ids = list(map(int, train_candidate_df.candidate_id.values))[0:10]
 
```

Comment:
Looks like you have some hard coded values here.  Will this stay fixed in the future?

Reply:
Ack! Good catch that was old code that needed to be changed